tutorials/arvo/arvo.md

```diff
@@ -0,0 +1,508 @@
+
+Contrast this with other popular operating systems, such as Windows, Mac, or Linux. The state of the operating system is something that crucially depends on having a constant power supply because much of the state of the operating system is stored in RAM, which is volatile. When you reboot your computer or suddenly lose power, any information stored in RAM is lost. Modern operating systems do mitigate this loss of information to some extent. For instance, it may remember what applications you were running at the time power was lost and try to restore them. Particularly durable programs may go as far as writing their state to disk every few seconds so that only very minimal information can be lost in a power outage. However, this is not the default behavior, and indeed if it were then they would be so slow as to be unusable, as you would effectively be using your hard disk as the RAM.
+
+How Arvo handles loss of power is closer to that of an SSD. Since it is an [ACID database](#acid-database) and a [single-level store](#single-level-store), a sudden loss of power have no effect on the state of the operating system. When you boot your ship back up it will be exactly as it was before the failure. Your information can never be lost, and because it was designed from the ground up to behave in this fashion, it does not suffer significant slowdown by persisting all data in this manner as would be the case if your typical operating system utilized its hard disk as RAM.
+
+
+### Over-the-air updates
+
+Arvo can hotpatch any other semantics at any layer in the system (apps, vanes, Arvo or Hoon itself) with automatic over-the-air updates.
+
+Typically, updates to an operating system are given via a pre-compiled binary,
+which is why some updates will work on some systems but not on others where the
+hardware and environment may differ. This is not so on Arvo - because it is an
+[interpreter](#solid-state-interpreter), Arvo may update itself by receiving
+source code from your sponsor over [Ames](@/docs/tutorials/arvo/ames.md), our
+network. As Hoon compiles down to Nock, which is an axiomatic representation of
+a deterministic computer, this code is guaranteed to run identically on your machine as it
+would on anybody else's.
+
+Since OTA updates can potentially alter the type system they must be performed
+outside of the type system. An OTA update contains a raw untyped Nock formula that
+encodes the new kernel as well as instructions on how to turn the final state of
+kernel into the initial state of the new kernel, which potentially includes things such as
+transforming the old type system into the new type system.
```

Comment:
Hmmm this seems to contradict what @joemfb told me last week, but I could be misremembering. Care to weigh in?

Reply:
@philipcmonk is right, with one additional detail: the transition from an old to a new kernel is untyped from the perspective the old kernel, but typed from the perspective of the new kernel. Since the transition can include a change to the type of type, it definitionally cannot be entirely performed within a type system (unless you posit a meta-typesystem, in which case I wish you luck ...).

Reply:
https://github.com/urbit/docs/pull/780/commits/bd126ddc5266ecbf21f147cf708f421acafcac00 @philipcmonk does this sound better?

Reply:
yep